server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -85,12 +88,14 @@ public SecretDAO(
       SecretContentDAOFactory secretContentDAOFactory,
       SecretSeriesDAOFactory secretSeriesDAOFactory,
       GroupDAO.GroupDAOFactory groupDAOFactory,
-      ContentCryptographer cryptographer) {
+      ContentCryptographer cryptographer,
+      PermissionCheck permissionCheck) {
     this.dslContext = dslContext;
     this.secretContentDAOFactory = secretContentDAOFactory;
     this.secretSeriesDAOFactory = secretSeriesDAOFactory;
     this.groupDAOFactory = groupDAOFactory;
     this.cryptographer = cryptographer;
+    this.permissionCheck = permissionCheck;
   }
```

Comment:
We don't use it here, but there's a fancy framework called Lombok that just allows you to slap something like @ Data or @ RequiredArgsConstructor on a class and boom! The constructor, getters, etc. are all auto-generated and you just have to add and remove fields (dependencies). Really cuts down on all this boilerplate.

https://projectlombok.org/

(just an FYI)